app/models/pipeline_run.rb

```diff
@@ -1405,4 +1408,43 @@ def status_url
                end
     base_url + "/samples/#{sample.id}/pipeline_runs"
   end
+
+  # Keys here are used as cache keys for report_info action in SamplesController.
+  # The values here are used as defaults for PipelineSampleReport.jsx.
+  def report_info_params
+    {
+      pipeline_version: pipeline_version || PipelineRun::PIPELINE_VERSION_WHEN_NULL,
+      # Default background is complicated... see get_background_id. In any case,
+      # we precache all backgrounds. See precache_report_info below.
+      background_id: nil,
+      # For invalidation when underlying data changes
+      report_ts: max_updated_at.utc.to_i,
+      format: "json"
+    }
+  end
+
+  # Gets last update time of all has_many relations and current pipeline run
+  def max_updated_at
+    assocs = PipelineRun.reflect_on_all_associations(:has_many)
+    assocs_max = assocs.map { |assoc| send(assoc.name).pluck(:updated_at).max }
+    [updated_at, assocs_max.compact.max].compact.max
+  end
+
+  # This precaches reports for *all* backgrounds. Each report took between 1 and
+  # 10s in testing.
+  def precache_report_info!
```

Comment:
If it is pre-caching for all Backgrounds, shouldn't it be a class method of Background model?

```
class Background
    def self.precache_all_ready_reports
```

Reply:
However, I might be wrong, but it feels it is not correct to pre-cache ALL ready Backgrounds. Shouldn't we be caching only data that is related to this specific pipeline_run?

Reply:
I chose to precache all background_id after all because Redis should act as a LRU cache when expires_in is enabled, so unused data will be evicted.

It's still not clear to me how the default background is selected and what cache hit rate would result. When @cdebourcy and I discussed this on the original PR, he recommended caching all backgrounds.

Reply:
In this logic, how do we know if Redis has already evicted the entry or not? I mean, I'm assuming this `ready: 1` is some sort of flag that represents if it is `ready` to be cached and it is turned to 0 when caching has been completed, is that correct? 
When redis evicts an entry, does it revert the flag back to `1`? Or are we reprocessing ALL backgrounds, every time when we invoke? This list seems to grow indefinitely, so there will be a day (maybe currently), where this query will return hundreds of thousands of lines, and we will be caching literally everything, defeating the purpose of an LRU cache.
Shouldn't we have a method to cache each Background individually? Shouldn't we invoke this method only when that data has been changed somehow? Shouldn't we do that only for things that are currently in use?

Reply:
And I also believe we should move this method to `Background` model. It seems to be unrelated to `PipelineRun`.

Reply:
Discussed offline. Greg clarified the relationship between a background and a pipeline_run.